include/exadg/solvers_and_preconditioners/multigrid/coarse_grid_solvers.h

```diff
@@ -197,98 +163,19 @@ class MGCoarseKrylov : public CoarseGridSolverBase<Operator>
 
     if(additional_data.preconditioner == MultigridCoarseGridPreconditioner::AMG)
     {
-      if(additional_data.amg_data.amg_type == AMGType::ML)
-      {
-#ifdef DEAL_II_WITH_TRILINOS
-        // create temporal vectors of type NumberAMG (double)
-        VectorTypeAMG dst_tri;
-        dst_tri.reinit(dst, false);
-        VectorTypeAMG src_tri;
-        src_tri.reinit(r, true);
-        src_tri.copy_locally_owned_data_from(r);
-
-        std::shared_ptr<PreconditionerML<Operator, NumberAMG>> coarse_operator =
-          std::dynamic_pointer_cast<PreconditionerML<Operator, NumberAMG>>(preconditioner_amg);
-
-        dealii::ReductionControl solver_control(additional_data.solver_data.max_iter,
-                                                additional_data.solver_data.abs_tol,
-                                                additional_data.solver_data.rel_tol);
-
-        if(additional_data.solver_type == KrylovSolverType::CG)
-        {
-          dealii::SolverCG<VectorTypeAMG> solver(solver_control);
-          solver.solve(coarse_operator->system_matrix, dst_tri, src_tri, *preconditioner_amg);
```

Comment:
here we use the `coarse_operator->system_matrix`.

Reply:
`coarse_operator` in the old code is probably a misleading name, since this object comes from a dynamic cast of the AMG preconditioner. So the system matrix of the AMG preconditioner is used for the vmult() in the Krylov solver, isn't it?

Reply:
that's what I meant, yes I think so too.